closingd/closing.c

```diff
@@ -530,18 +530,12 @@ int main(int argc, char *argv[])
 	/* Now we have first two points, we can init fee range. */
 	init_feerange(&feerange, commitment_fee, offer, deprecated_api);
 
-	/* Now apply the one constraint from above (other is inside loop). */
-	adjust_feerange(&cs, gossip_index, &channel_id, &feerange,
-			offer[!whose_turn], !whose_turn);
-
 	/* Now any extra rounds required. */
 	while (offer[LOCAL] != offer[REMOTE]) {
-		/* If they differ, adjust feerate. */
-		adjust_feerange(&cs, gossip_index, &channel_id, &feerange,
-				offer[whose_turn], whose_turn);
-
-		/* Now its the other side's turn. */
-		whose_turn = !whose_turn;
+		/* Still don't agree: adjust feerange based on previous offer */
+		adjust_feerange(&cs, gossip_index, &channel_id,
```

Comment:
It seems to me the below situation will erroneously pass?

0. `commitment_fee` = 1000
1. funder: 800
2. fundee: 200
3. funder: 900

`init_feerange` sets min 0, max 1000.  On entering the `while` loop we load the fundee fee into the feerange and set min 201, max 1000, but we do not constrain the funder side simllarly?

Reply:
Agreed.  Fix coming...

Reply:
Try those...